processing/src/main/java/org/apache/carbondata/processing/datatypes/PrimitiveDataType.java

```diff
@@ -412,16 +412,18 @@ private void updateValueToByteStream(DataOutputStream dataOutputStream, byte[] v
     dataOutputStream.write(value);
   }
 
-  private void updateNullValue(DataOutputStream dataOutputStream, BadRecordLogHolder logHolder)
-      throws IOException {
+  private void updateNullValue(Object input, DataOutputStream dataOutputStream,
+      BadRecordLogHolder logHolder) throws IOException {
     CarbonUtil.updateNullValueBasedOnDatatype(dataOutputStream, this.carbonDimension.getDataType());
-    String message = logHolder.getColumnMessageMap().get(carbonDimension.getColName());
-    if (null == message) {
-      message = CarbonDataProcessorUtil
-          .prepareFailureReason(carbonDimension.getColName(), carbonDimension.getDataType());
-      logHolder.getColumnMessageMap().put(carbonDimension.getColName(), message);
+    if (null == input) {
+      String message = logHolder.getColumnMessageMap().get(carbonDimension.getColName());
+      if (null == message) {
+        message = CarbonDataProcessorUtil
+            .prepareFailureReason(carbonDimension.getColName(), carbonDimension.getDataType());
+        logHolder.getColumnMessageMap().put(carbonDimension.getColName(), message);
+        logHolder.setReason(message);
```

Comment:
can move setReason to line 426 after closing one if case of (null == message)

Reply:
done